### DISCUSSION TITLE
Using both rancher and cluster autoscaler

### DISCUSSION THREAD

**Discussion by v-petukhov** [mysticaltech/terraform-hcloud-kube-hetzner#569] [Q&A]
Hi there!

Is there any option to use both rancher and cluster autoscaler? As I can see in `kube.tf` file notes, there are: 

  _Cluster Autoscaler
  Providing at least one map for the array enables the cluster autoscaler feature, default is disabled
  Please note that the autoscaler should not be used with initial_k3s_channel < "v1.25". **So ideally lock it to "v1.25".**_

And Rancher can not running using k3s v1.25, as I can understand, or I'm wrong with it? 

So, is there any option to use them both at one time?

Thanks for answers!

**Comment by valkenburg-prevue-ch** [comment]
You can use rancher v2.7.1 on kubernetes 1.25, which is what I do. Does the enable_rancher flag not use the latest rancher version yet? 

(I install rancher separately, outside of the kube.tf module.)

**Reply by v-petukhov** [reply]
As I can see from official [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-7-1/), there are no options to use Rancher 2.7.1 with k8s higher then 1.24

**Reply by valkenburg-prevue-ch** [reply]
Ah, sorry, you're absolutely right. Well then you have your answer unfortunately. However, it still depends on what functionality of rancher you use, as you can still just install it on 1.25, even if it is not officially supported.

**Reply by mysticaltech** [reply]
@v-petukhov I will remove these warnings from the docs, we use https://github.com/rancher/rancher/releases, and 2.7.1 is indeed the latest version. So just deploy everything without fixing the version and it will use the latest and greatest and everything should work!

**Reply by mysticaltech** [reply]
@v-petukhov Always assume that things work. The warning we give you is to get you on the safe side. Currently, the latest kube is v1.26, so we advise using v1.25 which is the default because it gets you maximum compatibility. BUT, you might as well find that rancher does work with v1.26 too. You would never know if you do not try! :)

**Reply by v-petukhov** [reply]
> @v-petukhov Always assume that things work. The warning we give you is to get you on the safe side. Currently, the latest kube is v1.26, so we advise using v1.25 which is the default because it gets you maximum compatibility. BUT, you might as well find that rancher does work with v1.26 too. You would never know if you do not try! :)

Thank you a lot! I'm really appreciate that! 

P.S. While I was waiting for any answers, I decided to try to install all the monitoring for the cluster manually (that's what I needed Rancher for) and my attempt was successful, thanks to this I received new and very interesting knowledge :)

**Reply by mysticaltech** [reply]
@v-petukhov Good to hear! We have a [show-and-tell](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/categories/show-and-tell) section in the discussions, if you feel like sharing your experience installing the monitoring solutions, it would be super useful for others (even a link would be great)! 🙏
